Khan/wonder-blocks
# Add component gallery descriptions

## Summary:
A component gallery was added in #2102 without descriptions.
In this PR, I added descriptions for all the listed components.

I also:
- added IDs to the the component headings so that the
  table of components scrolls to the sections correctly.
- added the external link icon to indicate that all
  the tiles link to their respective pages in new tabs.
- made the gallery more mobile-friendly

Issue: https://khanacademy.atlassian.net/browse/WB-1595

## Test plan:
- Go to http://localhost:6061/?path=/docs/overview--docs
- Confirm that the descriptions all make sense and are all in the same tense.
